src/block/block_compact.c

```diff
@@ -539,9 +537,17 @@ __compact_page_skip(
     else
         ++block->compact_pages_rewritten;
 
-    /* Estimate how much work is left. */
-    if (block->compact_pages_rewritten_expected == 0)
+    /*
+     * We must have reviewed at least some interesting number of pages for any estimates below to be
+     * worthwhile.
+     */
+    if (block->compact_pages_rewritten_expected == 0 &&
```

Comment:
You should check `compact_estimated` here instead of `compact_pages_rewritten_expected`.

Reply:
Done in 62b2b6a92d1ad8d2c861116e1d8703395e4ca7ad